File: python/23.05/80.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        idx = 0
        dic = collections.defaultdict(int)
        for i in range(len(nums)):
            dic[nums[i]] += 1
            if dic[nums[i]] > 2:
                continue
            else:
                nums[idx] = nums[i]
                idx += 1

        return idx

File: typescript/23.08/80.ts
function removeDuplicates(nums: number[]): number {
    let j = 0
    let visit = {}
    for (const num of nums) {
        if (!visit[num]) {
            visit[num] = 0
        }
        if (visit[num] < 2) {
            visit[num]++
            nums[j++] = num
        }
    }
    
    return j
};
